### DISCUSSION TITLE
How to deal with BelongsToMany relation?

### DISCUSSION THREAD

**Discussion by carlos-talavera** [abbasudo/laravel-purity#81] [Q&A]
Hi, this is a wonderful library and I've been using extensively on my project, but now I'm facing this issue and don't find how to do it right. I have this relation on a InsumoProduccion (ProductionInput) model to get the productive areas that the input belongs to:

```php
/**
 * Obtiene las áreas a las que pertenece el insumo de producción.
 *
 * @return \Illuminate\Database\Eloquent\Relations\BelongsToMany
 */
public function areas()
{
    return $this->belongsToMany(CatalogoArea::class, 'insumos_produccion_catalogo_areas', 'id_insumo_produccion', 'id_area');
}
```

I want to filter it based on the ID of the area. The corresponding model of the pivot table is this:

```php
class InsumoProduccionCatalogoArea extends Model
{
    use HasFactory, Filterable;

    protected $table = 'insumos_produccion_catalogo_areas';
    public $timestamps = false;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'id_insumo_produccion',
        'id_area'
    ];

    /**
     * The attributes that are filterable.
     *
     * @var array<string, string>
     */
    protected $filterFields = [
        'id_area'
    ];
}
```

I'm trying with something like this (in JS):

```js
areas: {
  id_area: {
    $eq: 6
  },
},
```

But it doesn't work, it brings all the records. I'd appreciate any idea or hint you could give me, thanks!

**Comment by abbasudo** [comment]
Hi @carlos-talavera. I am happy that purity came in handy for you.

you should define 
```php
    /**
     * The attributes that are filterable.
     *
     * @var array<string, string>
     */
    protected $filterFields = [
        'id_area'
    ];
```
in `CatalogoArea` class, not the intermediate class.

**Comment by carlos-talavera** [comment]
Hi @abbasudo, thanks for your answer. Unfortunately it doesn't work because the relation is like this

![imagen_2024-07-19_094225286](https://github.com/user-attachments/assets/a32700d9-36ee-4ce8-9110-3b66b452c005)

But if I try filtering by using `id` on `CatalogoArea`, it says that it's ambiguous. I tried specifying the table on the query using `catalogo_areas.id`, but it doesn't work neither. I have solved it as now by doing

```php
->when(isset($request->query('filters')['areas']['id_area']['$eq']), function ($query) use ($request) {
  $id_area = $request->query('filters')['areas']['id_area']['$eq'];

  $query->whereHas('areas', function ($query) use ($id_area) {
      $query->where('id_area', $id_area);
  });
})
```
Just to build the query as if I was using Purity
